src/dict.c

```diff
@@ -870,136 +869,163 @@ void dictTwoPhaseUnlinkFree(dict *d, dictEntry *he, dictEntry **plink, int table
     dictResumeRehashing(d);
 }
 
+#define DICT_SET_KEY(e, k)                                                                                             \
+    {                                                                                                                  \
+        if (entryIsNormal(e)) {                                                                                        \
+            _dictEntryNormal *_de = decodeEntryNormal(e);                                                              \
+            _de->key = k;                                                                                              \
+        } else if (entryIsNoValue(e)) {                                                                                \
+            _dictEntryNoValue *_de = decodeEntryNoValue(e);                                                            \
+            _de->key = k;                                                                                              \
+        } else {                                                                                                       \
+            assert(0);                                                                                                 \
+        }                                                                                                              \
+    }
+
+#define DICT_SET_VALUE(e, f, v)                                                                                        \
```

Comment:
I know `f` is just a placeholder for the value field. But what does it stand for? Could we be more creative with naming? Do we want add some code comment for each of these macro(s).

Reply:
sure - will expand the names. I was trying to keep the macros small.

Reply:
Hmm, the expansion is actually not easy as I can't use the same name as the real field name. I think `e`, `k`, and `v` are self-explanatory enough and we have been using single letter variables in the code consistently when their meaning is clear from the context (such as in this case). Nevertheless, I will add code comments to explain them at the top of this section.

Reply:
I don't think e is clear because we use de very consistently.

Reply:
I'll also throw in that when I re-read this comment I didn't immediately know what f was, which makes me think it's a bad name. K for key and v for value feel a lot more defensible.

Reply:
so `e -> de`, `f -> fld` but keep `k` and `v`?

Reply:
> so e -> de, f -> fld but keep k and v?

I'm fine with that.